### PR TITLE
Propagate registry mirror config for Nutanix provider

### PR DESCRIPTION
**Pull request by thunderboltsid**
To support EKS-Anywhere on Nutanix in air-gapped environments, we need to honor the
local registry mirror specified in the cluster config and propagate it forward.

**How has this been tested?**
Followed https://anywhere.eks.amazonaws.com/docs/reference/clusterspec/optional/registrymirror/ to set up a
harbor registry with necessary projects and importing the images to the harbor registry.
```
$ bin/eksctl-anywhere import images -i eks-anywhere-images.tar --bundles eks-anywhere-downloads/bundle-release.yaml --registry ${REGISTRY_ENDPOINT}
```
Add the `registryMirrorConfiguration` property to the cluster spec pointing to the new registry.
```
apiVersion: anywhere.eks.amazonaws.com/v1alpha1
kind: Cluster
metadata:
  name: eksa-sid
spec:
  ...
  registryMirrorConfiguration:
    endpoint: registry.ntnxsherlock.com
    ociNamespaces:
      - registry: "public.ecr.aws/l0g8r8j6"
        namespace: "eks-anywhere"
    authenticate: true
    insecureSkipVerify: true
```
Using a bundle-release.yaml with an image of the controller built from the changes in this PR as bundles override, create a cluster.
```
$ bin/eksctl-anywhere create cluster -f ./${CLUSTER_NAME}.yaml --bundles-override ./bundle-release.yaml
Logging in to docker registry registry.ntnxsherlock.com
Logging in to docker registry registry.ntnxsherlock.com
Logging in to docker registry registry.ntnxsherlock.com
Logging in to docker registry registry.ntnxsherlock.com
Logging in to docker registry registry.ntnxsherlock.com
Logging in to docker registry registry.ntnxsherlock.com
Logging in to docker registry registry.ntnxsherlock.com
Logging in to docker registry registry.ntnxsherlock.com
Logging in to docker registry registry.ntnxsherlock.com
Logging in to docker registry registry.ntnxsherlock.com
Logging in to docker registry registry.ntnxsherlock.com
Logging in to docker registry registry.ntnxsherlock.com
Logging in to docker registry registry.ntnxsherlock.com
Logging in to docker registry registry.ntnxsherlock.com
Logging in to docker registry registry.ntnxsherlock.com
Logging in to docker registry registry.ntnxsherlock.com
Logging in to docker registry registry.ntnxsherlock.com
Logging in to docker registry registry.ntnxsherlock.com
Performing setup and validations
✅ Nutanix Provider setup is valid
✅ Validate certificate for registry mirror
✅ Validate authentication for git provider
Creating new bootstrap cluster
Provider specific pre-capi-install-setup on bootstrap cluster
Installing cluster-api providers on bootstrap cluster
Provider specific post-setup
Creating new workload cluster
Installing networking on workload cluster
Logging in to helm registry	{"registry": "registry.ntnxsherlock.com:443"}
Creating EKS-A namespace
Installing cluster-api providers on workload cluster
Installing EKS-A secrets on workload cluster
Installing resources on management cluster
Moving cluster management from bootstrap to workload cluster
Installing EKS-A custom components (CRD and controller) on workload cluster
Installing EKS-D components on workload cluster
Creating EKS-A CRDs instances on workload cluster
Installing GitOps Toolkit on workload cluster
GitOps field not specified, bootstrap flux skipped
Writing cluster config file
Deleting bootstrap cluster
🎉 Cluster created!
```

### CODE REVIEW

**Comment by thunderboltsid** [comment]
> Can support for the `InsecureSkipVerify` flag in `RegistryMirrorConfiguration` be included as well?
> 
> [Here](https://github.com/aws/eks-anywhere/issues/1647) is the related issue describing the efforts to propagate it across all providers. You can also reference [this](https://github.com/aws/eks-anywhere/pull/5235) PR

Sure.

**Comment by thunderboltsid** [comment]
@cxbrowne1207 Updated the PR with support for propagating of skipping TLS verification in the templates. Can you please take a look?

**Review by cxbrowne1207 — COMMENTED** [review]
/lgtm

File: pkg/api/v1alpha1/cluster.go [hunk 1]
```diff
@@ -724,9 +724,13 @@ func validateMirrorConfig(clusterConfig *Cluster) error {
...
-	if !features.RegistryMirrorInsecureSkipVerifySupport().IsActive() {
-		if clusterConfig.Spec.RegistryMirrorConfiguration.InsecureSkipVerify && clusterConfig.Spec.DatacenterRef.Kind != SnowDatacenterKind {
-			return errors.New("insecureSkipVerify is only supported for snow provider")
+	if !features.RegistryMirrorInsecureSkipVerifySupport().IsActive() &&
+		clusterConfig.Spec.RegistryMirrorConfiguration.InsecureSkipVerify {
+		switch clusterConfig.Spec.DatacenterRef.Kind {
+		case NutanixDatacenterKind, SnowDatacenterKind:
+			break
+		default:
+			return errors.New("insecureSkipVerify is only supported for nutanix and snow providers")
```
Comment (cxbrowne1207) [inline pkg/api/v1alpha1/cluster.go:733]: Can we just use the feature flag `REGISTRY_MIRROR_INSECURE_SKIP_VERIFY_SUPPORT` here to bypass the validation for now? I think eventually we could remove it altogether validation once it's successfully supported for all the other providers.

Comment (thunderboltsid) [reply pkg/api/v1alpha1/cluster.go:733]: Yeah, until it's supported for all providers, this is needed.

Comment (cxbrowne1207) [reply pkg/api/v1alpha1/cluster.go:733]: I think we need to remove the feature flag check as well then. The purpose of it was to have support for snow only, unless that flag was set.

File: test/e2e/nutanix_test.go [hunk 2]
```diff
@@ -775,3 +776,41 @@ func TestNutanixKubernetes126OIDC(t *testing.T) {
 	)
 	runOIDCFlow(test)
 }
+
+// Registry Mirror Tests
+func TestNutanixKubernetes124UbuntuRegistryMirrorAndCert(t *testing.T) {
```
Comment (abhinavmpandey08) [inline test/e2e/nutanix_test.go:781]: do you mind moving the e2e to a separate PR? We will need to setup some infra for this which we don't have currently.

**Review by abhinavmpandey08 — APPROVED** [review]
/approve
/lgtm
/woof
